### PR TITLE
Ensure all `StorageVersion`s on Polkadot/Kusama are correct

### PR DESCRIPTION
**Pull request by bkchr**
With the recent change in Substrate we now detect issues with `StorageVersion` when running `try-runtime`. This pr fixes all the issues to make `try-runtime` working for Polkadot/Kusama again.

### CODE REVIEW

File: xcm/pallet-xcm/src/migration.rs [hunk 1]
```diff
@@ -51,7 +49,7 @@ pub mod v1 {
 				VersionNotifyTargets::<T>::translate_values(translate);
 
 				log::info!("v1 applied successfully");
-				STORAGE_VERSION.put::<Pallet<T>>();
+				StorageVersion::new(1).put::<Pallet<T>>();
```
Comment (ggwpez) [inline xcm/pallet-xcm/src/migration.rs:52]: I think we should pay attention in general to never use `STORAGE_VERSION.put`.  
There is one more case in `runtime/parachains/src/configuration/migration.rs`.

**Review by ggwpez — APPROVED** [review]
Noice. Lets see what the CI says.

Now we just need https://github.com/paritytech/substrate/pull/13013 and then we are set.

**Comment by kianenigma** [comment]
Doesn't this address parts of https://github.com/paritytech/polkadot/issues/7187?

File: xcm/pallet-xcm/src/migration.rs (hunk 1 above)
Comment (kianenigma) [reply xcm/pallet-xcm/src/migration.rs:52]: > I think we should pay attention in general to never use STORAGE_VERSION.put.

Why?

**Comment by bkchr** [comment]
> Doesn't this address parts of #7187?

Yes

File: xcm/pallet-xcm/src/migration.rs (hunk 1 above)
Comment (bkchr) [reply xcm/pallet-xcm/src/migration.rs:52]: Because when you change `STORAGE_VERSION`, you will change the version the migration is writing to storage.

**Review by sam0x17 — APPROVED** [review]

**Review by liamaharon — APPROVED** [review]

**Review by ordian — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

**Review by gavofyork — APPROVED** [review]

**Comment by liamaharon** [comment]
Tested `on_runtime_upgrade` with a runtime built from this branch, looks good.

`try-runtime-cli` `on-runtime-upgrade` is currently broken when used with polkadot runtimes built from https://github.com/paritytech/polkadot/commit/1203b2519fed1727256556fb879c6c03c27a830d and above, this fixes it so going to merge it.

**Pull request merged**